src/plugins/data/public/query/query_service.ts

```diff
@@ -81,13 +82,21 @@ export class QueryService implements PersistableStateService<QueryState> {
 
   state$!: QueryState$;
 
-  public setup({ storage, uiSettings, nowProvider }: QueryServiceSetupDependencies): QuerySetup {
+  constructor(private minRefreshInterval: number = 5000) {}
+
+  public setup({
+    storage,
+    uiSettings,
+    nowProvider,
+    minRefreshInterval = this.minRefreshInterval,
+  }: QueryServiceSetupDependencies): QuerySetup {
```

Comment:
Nit: Do we need `minRefreshInterval` passed through `setup` here, or can we just read it from `this.minRefreshInterval`?

Reply:
The `security_solution` plugin is using two custom instances of the `QueryService`, so it makes it easier to set the `minRefreshInterval` allowing it in the `setup` call as well as the constructor.

https://github.com/elastic/kibana/blob/ce93360add5227f378d0194993f775f066a74430/x-pack/plugins/security_solution/public/plugin_services.ts#L76-L88